src/jobs/microk8s_install.yml

```diff
@@ -63,18 +72,21 @@ steps:
         # install microk8s and selected addons
         sudo snap install microk8s --classic --channel << parameters.microk8s_channel >>
         export PATH=$PATH:/snap/bin
-        sudo microk8s.start && sudo microk8s.status --wait-ready
-        sudo microk8s.enable ingress dns storage && sudo microk8s.status --wait-ready
+        sudo microk8s.start && sudo microk8s.status --wait-ready --timeout 180
+
+        # enabling plugins one at a time to prevent a race condition
+        # see https://github.com/ubuntu/microk8s/issues/706
+        sudo microk8s.enable dns && sleep 10 && sudo microk8s.status --wait-ready --timeout 180
```

Comment:
all those sleeps are extremely suspicious.
Please explain why we are doing them, why they are necessary and place big comments around, or better, find another way that doesn't require sleeps :-)

Reply:
I know... :-(  I'm definitely not proud of it... I tried to find a better way to do it but nothing else worked. `microk8s.enable` returns, as if it was ready for requests, even though it isn't, and then crashes saying that microk8s is not running when we run `microk8s.status`. This is flaky in microk8s. See, for instance, this `https://github.com/ubuntu/microk8s/issues/375`

The sleep fixes the issue. I'd like to keep it, for now, so that we can move forward with this. It is really ugly and it bothers me, so I definitely want to change it to something better when I find a way. I'll make better comments for this so that it doesn't slip under the radar.